src/core/thread/address_resolver.cpp

```diff
@@ -104,7 +104,7 @@ void AddressResolver::Remove(uint8_t routerId)
     {
         if (Mle::Mle::GetRouterId(mCache[i].mRloc16) == routerId)
         {
-            InvalidateCacheEntry(mCache[i]);
+            InvalidateCacheEntry(mCache[i], "removed routerId");
```

Comment:
Will these strings get compiled out when logging is disabled?

Reply:
I guess may depends on optimization capability level of compiler...

Reply:
Will change them to enum constant which we can then convert to string when adding the log...